src/main/java/com/exadel/movie_new/service/MovieServiceImpl.java

```diff
@@ -28,19 +36,29 @@ public Movie addMovie(Movie movie) {
     @Override
     public List<Movie> getAllMovies() {
         List<Movie> movieData = readJsonData();
+        if (movieData.isEmpty()) {
+            throw new MovieNotFoundException("no movies found");
+        }
         return movieData;
     }
 
-
     @Override
     public Movie getMovie(String movieId) {
         List<Movie> movieData = readJsonData();
-        return movieData.parallelStream().filter(mov -> mov.getId().equals(movieId)).findAny().orElse(null);
+        Movie searchedMovie = movieData.parallelStream().filter(mov -> mov.getId().equals(movieId)).findAny().orElse(null);
+        if (searchedMovie == null) {
+            throw new MovieNotFoundException("requested movie not found");
+        }
+        return searchedMovie;
     }
 
     @Override
     public Movie updateMovie(Movie movie) {
         List<Movie> movieData = readJsonData();
+        Movie searchedMovie = movieData.parallelStream().filter(mov -> mov.getId().equals(movie.getId())).findAny().orElse(null);

```

Comment:
if you use the same code twice - it's better to move it in separate method and reuse it any time you need